### DISCUSSION TITLE
Prevent 'cli' prompt being displayed on main thread.

### DISCUSSION THREAD

**Discussion by patrickn** [daniele77/cli#134] [Q&A] [answered]
Hi and thanks for CLI, it looks really interesting.

I am just starting out and going through the examples. I want to create the scheduler and telnet server on the main thread and then run scheduler.Run() in a background thread. Is there a way to prevent the 'cli' prompt from being displayed on the main thread.

Thanks.

**Comment by daniele77** [comment, marked as answer]
Well, a stream is not displayed on "a thread", it's displayed on "a console".
If you don't want the standard console output but only the remote console (i.e., telnet), you simply don't instantiate `CliLocalTerminalSession`.
E.g.:

        MainScheduler scheduler;
        // local session not needed
        /*
        CliLocalTerminalSession localSession(cli, scheduler, std::cout, 200);
        localSession.ExitAction(
            [&scheduler](auto& out) // session exit action
            {
                out << "Closing App...\n";
                scheduler.Stop();
            }
        );
        */

        // setup server

        CliTelnetServer server(cli, scheduler, 5000);
        // exit action for all the connections
        server.ExitAction( [](auto& out) { out << "Terminating this session...\n"; } );

        scheduler.Run();

**Reply by patrickn** [reply]
Doh, I don't know how I missed that? Thanks for the reply. :-)

**Comment by daniele77** [comment]
Don't worry. `Github Discussions` exists to make up for missing documentation :-)

If you want, click on the star button on top of this page (or -- even better -- click on the heart button!!!)
